src/meta/Dimensions.ts

```diff
@@ -0,0 +1,63 @@
+import { Base } from './Base';
+
+export interface DimensionResults {
+	bottom: number;
+	height: number;
+	left: number;
+	offsetHeight: number;
+	offsetLeft: number;
+	offsetTop: number;
+	offsetWidth: number;
+	right: number ;
+	scrollHeight: number;
+	scrollLeft: number;
+	scrollTop: number;
+	scrollWidth: number;
+	top: number;
+	width: number;
+}
+
+export class Dimensions extends Base {
```

Comment:
A really minor nit, but do you think this would be a little "nicer" with something like this?

```ts
const defaultDimensions = {
	bottom: 0,
	height: 0,
	left: 0,
	right: 0,
	top: 0,
	width: 0,
	scrollLeft: 0,
	scrollTop: 0,
	scrollHeight: 0,
	scrollWidth: 0,
	offsetLeft: 0,
	offsetTop: 0,
	offsetWidth: 0,
	offsetHeight: 0
};

export class Dimensions extends Base {
	public get(key: string): Readonly<DimensionResults> {
		this.requireNode(key);
		const node = nodes.get(key);
		if (!node) { 
			return { ...defaultDimensions };
		}
		const boundingDimensions = node.getBoundingClientRect();

		return {
			scrollLeft: node.scrollLeft,
			scrollTop: node.scrollTop,
			scrollHeight: node.scrollHeight,
			scrollWidth: node.scrollWidth,
			offsetLeft: node.offsetLeft,
			offsetTop: node.offsetTop,
			offsetWidth: node.offsetWidth,
			offsetHeigh: node.offsetHeight,
			bottom: boundingDimensions.bottom,
			height: boundingDimensions.height,
			left: boundingDimensions.left,
			right: boundingDimensions.right,
			top: boundingDimensions.top,
			width: boundingDimensions.width,
		};
	}
}
```